src/components/Dialogs/LockProvider.tsx

```diff
@@ -506,7 +538,10 @@ export const LockProvider = ({
       transactions.push("deploy_lockup");
     }
 
-    if (selectedToken?.type === "lst" && !stakingPoolId) {
+    if (
+      selectedToken?.type === "lst" &&
+      (!stakingPoolId || stakingPoolId !== selectedToken.accountId)
```

Comment:
I don't think we can simply check if `stakingPoolId !== selectedToken.accountId` unfortunately. 

In order to switch pools you must also NOT have an active balance with that pool i.e. you must unstake/withdraw the funds from the pool if it's a legacy staking pool OR withdraw the funds if it's a LST. The latter is not possible with the current contracts (you can read more about why that is [here](https://gov.houseofstake.org/info?item=fungible-token-withdrawal)).